repo/fsrepo/fsrepo.go

```diff
@@ -102,6 +112,24 @@ func open(repoPath string) (repo.Repo, error) {
 	if !isInitializedUnsynced(r.path) {
 		return nil, debugerror.New("ipfs not initialized, please run 'ipfs init'")
 	}
+
+	// Check version, and error out if not matching
+	ver, err := ioutil.ReadFile(path.Join(expPath, "version"))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, errors.New("version check failed, no version file found, please run 0-to-1 migration tool.")
+		}
+		return nil, err
+	}
+
+	vers := string(ver)[:1]
+
+	if vers != RepoVersion {
+		return nil, fmt.Errorf("Repo has incorrect version: '%s'\nProgram version is: '%s'\nPlease run the appropriate migration tool before continuing",
```

Comment:
we could at least just have a page somewhere discussing migrations, maybe a github repo, and link them to it in the error message